msu/hooks/skills/skill_container.nut

```diff
@@ -27,6 +27,74 @@
 		this.m.ScheduledChangesSkills.clear();
 	}
 
+	local add = o.add;
+	o.add = function( _skill, _order = 0 )
+	{
+		if (!_skill.isStacking())
+		{
+			local incrementAddedStack = function( _alreadyPresentSkill, _skillToAdd )
+			{
+				_alreadyPresentSkill.m.MSU.AddedStack += 1;
+				if (!::MSU.isNull(_skillToAdd.getItem()))
+				{
+					if (::MSU.isNull(_alreadyPresentSkill.getItem()) _alreadyPresentSkill.setItem(_skillToAdd.getItem());
+					foreach (i, itemSkill in _skillToAdd.getItem().m.SkillPtrs)
+					{
+						if (itemSkill.getID() == _skillToAdd.getID())
+						{
+							_skillToAdd.getItem().m.SkillPtrs[i] = _alreadyPresentSkill;
+							_skillToAdd.setItem(null);
+							break;
+						}
+					}
+				}
+
+				if (_alreadyPresentSkill.m.MSU.AddedStack > 0)
+				{
+					_alreadyPresentSkill.m.IsGarbage = false;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.Skills)
+			{
+				if (alreadyPresentSkill.getID() == _skill.getID())
+				{
+					incrementAddedStack(alreadyPresentSkill, _skill);
+					if (alreadyPresentSkill.m.MSU.AddedStack > 1) alreadyPresentSkill.onRefresh();
+					return;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.SkillsToAdd)
```

Comment:
I don't know how these work, but by the name they make me think that they'll be `add()`-ed at their own time, so then this would increment twice maybe?

Reply:
Nah, they just get removed from the `this.m.SkillsToAdd` array and pushed to `this.m.Skills` array during the `collectGarbage` function. No new call to `add` function.